AppWidget/glance-widget/src/main/java/com/example/android/glancewidget/TodoListGlanceWidget.kt

```diff
@@ -0,0 +1,157 @@
+import androidx.glance.text.FontWeight
+import androidx.glance.text.Text
+import androidx.glance.text.TextStyle
+
+private val groceryStringIds = listOf(
+    R.string.grocery_list_milk,
+    R.string.grocery_list_eggs,
+    R.string.grocery_list_tomatoes,
+    R.string.grocery_list_bacon,
+    R.string.grocery_list_butter,
+    R.string.grocery_list_cheese,
+    R.string.grocery_list_potatoes,
+    R.string.grocery_list_broccoli,
+    R.string.grocery_list_salmon,
+    R.string.grocery_list_yogurt
+)
+
+/**
+ * Glance widget that showcases how to use:
+ * - Compound buttons
+ * - LazyColumn
+ * - State management using GlanceStateDefinition
+ */
+class TodoListGlanceWidget : GlanceAppWidget() {
+
+    override var stateDefinition: GlanceStateDefinition<*> = PreferencesGlanceStateDefinition
+
+    @Composable
+    override fun Content() {
+        Column(
+            modifier = GlanceModifier
+                .fillMaxSize()
+                .background(color = Color.White)
+                .padding(8.dp)
+        ) {
+            val context = LocalContext.current
+            val prefs = currentState<Preferences>()
+            Text(
+                text = context.getString(R.string.glance_todo_list),
+                modifier = GlanceModifier
+                    .fillMaxWidth()
+                    .padding(8.dp),
+                style = TextStyle(fontWeight = FontWeight.Bold, fontSize = 18.sp),
+            )
+            CountToggled()
+            LazyColumn {
+                items(groceryStringIds) {
+                    val toggledString = context.getString(it)
+                    val checked = prefs[booleanPreferencesKey(toggledString)] ?: false
+                    CheckBox(
+                        text = toggledString,
+                        checked = prefs[booleanPreferencesKey(toggledString)] ?: false,
+                        onCheckedChange = actionRunCallback<CheckboxClickAction>(
+                            actionParametersOf(
+                                toggledKey to toggledString,
+                                toggledValue to !checked
+                            )
+                        ),
+                        modifier = GlanceModifier.padding(12.dp)
+                    )
+                }
+            }
+        }
+    }
+}
+
+private val toggledKey = ActionParameters.Key<String>("ToggledKey")
+private val toggledValue = ActionParameters.Key<Boolean>("ToggledValue")
+
+class CheckboxClickAction : ActionCallback {
+    override suspend fun onRun(context: Context, glanceId: GlanceId, parameters: ActionParameters) {
+        val toggledString = requireNotNull(parameters[toggledKey]) {
+            "Add $toggledKey parameter in the ActionParameters."
+        }
+        // TODO: Remove this parameter once the checked state can be retrieved through
+        // ToggleableStateKey.
+        val checked = requireNotNull(parameters[toggledValue]) {
```

Comment:
You should use `ToggleableStateKey` as key in the parameters object to get the checked / unchecked value.

Reply:
As I commented in another thread, I thought ToggleableStateKey was not working in a lazy list, but done.

Reply:
You can see in the demos of glance that ScrollableAppWidget uses this.